### PR TITLE
feat: implement WithConfig client option

### PR DESCRIPTION
**Pull request by QuentinBrosse**
- Implement a `func WithConfig(config scwconfig.Config) ClientOption` to configure a client with Scaleway configuration.
- Move `scw.{Region|Zone}` to `utils` package.

### CODE REVIEW

File: scw/client_option.go [hunk 1]
```diff
@@ -51,6 +52,42 @@ func WithHTTPClient(httpClient *http.Client) ClientOption {
 	}
 }
 
+// WithConfig client option configure a client with Scaleway configuration.
+func WithConfig(config scwconfig.Config) ClientOption {
+	return func(s *settings) {
+		accessKey, accessKeyExist := config.GetAccessKey()
+		secretKey, secretKeyExist := config.GetSecretKey()
+		if accessKeyExist && secretKeyExist {
```
Comment (kindermoumoute) [inline scw/client_option.go]: I know it changes the old behavior, but this should work from the moment the secret key exists.

File: scw/client_test.go [hunk 2]
```diff
@@ -11,11 +11,12 @@ import (
 const (
 	testAPIURL                = "https://api.example.com/"
 	defaultAPIURL             = "https://api.scaleway.com"
-	testAccessKey             = "some access key"
-	testSecretKey             = "some secret key"
-	testDefaultOrganizationID = "some default organization id"
+	testAccessKey             = "ACCESS_KEY"
+	testSecretKey             = "SECRET_KEY"
```
Comment (kindermoumoute) [inline scw/client_test.go]: Secret key must be uuid (in case we add validation later)

File: scw/client_test.go [hunk 3]
```diff
@@ -33,29 +34,74 @@ func TestNewClientWithDefaults(t *testing.T) {
...
 
-	testhelpers.Equals(t, testAPIURL, client.apiURL)
-	testhelpers.Equals(t, auth.NewToken(testAccessKey, testSecretKey), client.auth)
+		testhelpers.Equals(t, auth.NewToken(testAccessKey, testSecretKey), client.auth)
+		testhelpers.Equals(t, testAPIURL, client.apiURL)
 
-	testhelpers.Equals(t, someHTTPClient, client.httpClient)
+		clientTransport, ok := client.httpClient.Transport.(*http.Transport)
+		testhelpers.Equals(t, true, ok)
+		testhelpers.NotNil(t, clientTransport.TLSClientConfig)
```
Comment (kindermoumoute) [inline scw/client_test.go]: Use testhelpers.Assert instead

File: internal/testhelpers/test_helpers.go [hunk 4]
```diff
@@ -34,3 +34,12 @@ func Equals(tb testing.TB, exp, act interface{}) {
 		tb.FailNow()
 	}
 }
+
+// NotNil check if val is not nil.
+func NotNil(tb testing.TB, val interface{}) {
```
Comment (kindermoumoute) [inline internal/testhelpers/test_helpers.go]: This kind of test should be done with testhelpers.Assert

File: scw/client_test.go [hunk 5]
```diff
@@ -33,29 +34,74 @@ func TestNewClientWithDefaults(t *testing.T) {
...
+		testhelpers.Ok(t, err)
 
-	testhelpers.Equals(t, testAPIURL, client.apiURL)
-	testhelpers.Equals(t, auth.NewToken(testAccessKey, testSecretKey), client.auth)
+		testhelpers.Equals(t, auth.NewToken(testAccessKey, testSecretKey), client.auth)
+		testhelpers.Equals(t, testAPIURL, client.apiURL)
 
-	testhelpers.Equals(t, someHTTPClient, client.httpClient)
+		clientTransport, ok := client.httpClient.Transport.(*http.Transport)
+		testhelpers.Equals(t, true, ok)
```
Comment (kindermoumoute) [inline scw/client_test.go]: Could switch to testhelpers.Assert here

File: scw/client_option.go (hunk 1 above)
Comment (QuentinBrosse) [reply scw/client_option.go]: You speak about a case in which user compose with `WithAuth` (in which `accessKey` AND `sercretKey` are set) and `WithConfig` (in which `accessKey` OR `sercretKey` are set).

File: scw/client_option.go [hunk 6]
```diff
@@ -51,6 +52,42 @@ func WithHTTPClient(httpClient *http.Client) ClientOption {
...
+		if exist {
+			s.insecure = insecure
+		}
+
+		defaultOrganizationID, exist := config.GetDefaultOrganizationID()
+		if exist {
+			s.defaultOrganizationID = defaultOrganizationID
+		}
+
+		defaultRegion, exist := config.GetDefaultRegion()
```
Comment (QuentinBrosse) [inline scw/client_option.go]: Should return a `Region`

File: scw/client_option.go [hunk 7]
```diff
@@ -51,6 +52,42 @@ func WithHTTPClient(httpClient *http.Client) ClientOption {
...
+		if exist {
+			s.defaultRegion = Region(defaultRegion)
+		}
+
+		defaultZone, exist := config.GetDefaultZone()
```
Comment (QuentinBrosse) [inline scw/client_option.go]: Should return a `Zone`

File: README.md [hunk 8]
```diff
@@ -34,13 +34,20 @@ import (
 
 	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
 	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/scaleway-sdk-go/scwconfig"
 )
 
 func main() {
 
+	// Get Scaleway Config
+	config, err := scwconfig.Load() // Get your credentials at https://console.scaleway.com/account/credentials
```
Comment (kindermoumoute) [inline README.md]: Keep the previous example in the readme, but add this in in [example_test.go](https://github.com/scaleway/scaleway-sdk-go/blob/master/example_test.go) 

```
func Example_apiClientWithConfig() {
	...
}
```

File: example_test.go [hunk 9]
```diff
@@ -28,6 +29,28 @@ func Example_apiClient() {
 
 }
 
+func Example_apiClientWithConfig() {
+
+	// Get Scaleway Config
+	config, err := scwconfig.Load() // Get your credentials at https://console.scaleway.com/account/credentials
```
Comment (kindermoumoute) [inline example_test.go]: check err?

File: example_test.go [hunk 10]
```diff
@@ -27,6 +29,28 @@ func Example_apiClient() {
...
```
Comment (jerome-quere) [inline example_test.go]: Remove comment

File: scw/client_option.go [hunk 11]
```diff
@@ -55,9 +56,9 @@ func WithHTTPClient(httpClient *http.Client) ClientOption {
 // WithConfig client option configure a client with Scaleway configuration.
 func WithConfig(config scwconfig.Config) ClientOption {
 	return func(s *settings) {
-		accessKey, accessKeyExist := config.GetAccessKey()
+		accessKey, _ := config.GetAccessKey()
```
Comment (jerome-quere) [inline scw/client_option.go]: Add a comment explaining why you do not test accessKeyExist

File: scw/client_test.go [hunk 12]
```diff
@@ -6,16 +6,17 @@ import (
...
 	"github.com/scaleway/scaleway-sdk-go/internal/testhelpers"
+	"github.com/scaleway/scaleway-sdk-go/utils"
 )
 
 const (
 	testAPIURL                = "https://api.example.com/"
 	defaultAPIURL             = "https://api.scaleway.com"
 	testAccessKey             = "ACCESS_KEY"
-	testSecretKey             = "SECRET_KEY"
+	testSecretKey             = "539a6564-bf92-4dc9-a0d4-50e3ca827ecb"
```
Comment (jerome-quere) [inline scw/client_test.go]: put something like  "11111111-1111-1111-1111-111111111111"

**Pull request merged**